### DISCUSSION TITLE
萌新请教custom_cn_mark.txt和force_dnscrypt_list.txt的区别

### DISCUSSION THREAD

**Discussion by lich-un** [kkkgo/PaoPaoDNS#228] [问答和帮助] [answered]
首先感谢kkkgo大佬开发了这么好用的分流工具，昨天已经配合paopaogw搭建成功。

阅读讨论区中的其他相关内容，当遇到非大陆IP需要走直连的情况时，看到大佬通常推荐将域名维护到force_dnscrypt_list.txt中。
而对于custom_cn_mark.txt，虽然kkkgo大佬解释了“有限的使用场景：当域名被USE_MARK_DATA或者被IP库认定为非CN域名但你希望把他当成CN域名处理的时候。”和"实际适用场景: 域名是anycast地址，境内解析可信任， 子域名包含CDN，不方便直接全部加入force_dnscrypt_list.txt的情况。"
碍于萌新对网络知识的一知半解，还是不太理解这两个文件的用途区别在哪，或者当域名维护到这两个列表中时，软件的行为有什么区别？
具体来说对于讨论 #156 的情况，我看日期当时的版本还没有custom_cn_mark.txt这个文件，所以大佬推荐将苹果的域名加入到force_dnscrypt_list.txt中，但如果是最新的版本实际上这些域名是不是加入custom_cn_mark.txt中更好？因为“apple的大多数是anycast的IP”而custom_cn_mark.txt"实际适用场景: 域名是anycast地址..."

另外“不方便直接全部加入force_dnscrypt_list.txt的情况。”这句要怎么理解？

能不能请大佬再更详细的讲解一下？感谢

**Comment by kkkgo** [comment, marked as answer]
实际行为区别：
force_dnscrypt_list： 强制走dnscrypt加密解析。dnscrypt服务器都位于境外。
USE_MARK_DATA：具有境外域名数据和少量境内域名数据。行为如下：
 - 当域名命中境内域名数据时跳过读取境外域名数据。
 - 当域名命中境外域名数据的时候会提前按境外域名流程处理（而一般流程是先尝试递归或者国内转发，再把结果用IP库判断处理）。
 - 当域名命中境内域名数据时，无论递归还是转发有没有结果，都会提前终止解析流程（而一般流程是回落到境外域名流程处理，比如走dnscrypt加密解析）。
 
custom_cn_mark：启用USE_MARK_DATA的时候生效，当附加境内域名数据处理，等效于USE_MARK_DATA中的境内数据。

实际效果区别：
force_dnscrypt_list：可以保证结果不被污染，可以保证结果从境外解析（DNS不泄漏），解析结果为真实结果（区别于FAKEIP场景下）。
custom_cn_mark：等效于USE_MARK_DATA中的境内数据，不能保证结果不被污染，可以保证结果从境内解析（DNS泄漏），解析结果为真实结果（区别于FAKEIP场景下）。

实际优先级：force_* > USE_MARK_DATA

**Reply by lich-un** [reply]
感谢详尽的说明，现在明白了（我觉得我明白了。。。）
所以如果是从steam、ps5游戏机或者苹果app store上下载的场景（越快越好，且内容不敏感？）应该把域名加到custom_cn_mark里更好，而不是force_dnscrypt_list。我这样理解对吗？

比如我打开ps5游戏机的时候，从ui的连接页里观察到了多个*.playstation.com或*.playstation.net的子域名，那我是不是应该在custom_cn_mark里添加以下两行？
domain:playstation.com
domain:playstation.net

再比如在讨论 #47 中，提到的下边3行steam下载相关的，是不是也放在custom_cn_mark是最好的选择？
domain:steamcontent.com
domain:akamaized.net
domain:akamaihd.net

**Reply by kkkgo** [reply]
取决于你实际场景需求和体验，比如有的地方playstation的解析可能有污染，或者部分playstation域名走代理体验更好。

**Reply by lich-un** [reply]
理解了，感谢耐心解答
